### PR TITLE
Add select_nodes_for_role helper to help create default proposals

### PR DESCRIPTION
**Pull request by vuntz**
This will select nodes for roles, based on constraints.

### CODE REVIEW

File: crowbar_framework/app/models/service_object.rb [hunk 1]
```diff
@@ -649,6 +645,33 @@ def find_dep_proposal(bc, optional=false)
...
+    violates_admin_constraint?(elements, role) ||
+      violates_platform_constraint?(elements, role) ||
+      violates_exclude_platform_constraint?(elements, role) ||
+      violates_cluster_constraint?(elements, role)
+  end
+
+  # Helper to select nodes that make sense on proposal creation
+  def select_nodes_for_role(all_nodes, role, preferred_intended_role = nil)
+    # do not modify array given by caller
+    all_nodes = all_nodes.dup
```
Comment (jsuchome) [inline crowbar_framework/app/models/service_object.rb]: Just a nitpicking - what about different name than all_nodes, when it's actually be a subset anyway...

**Comment by tboerger** [comment]
LGTM, good step in the correct direction

File: crowbar_framework/app/models/service_object.rb (hunk 1 above)
Comment (vuntz) [inline crowbar_framework/app/models/service_object.rb]: Good point, fixed.

**Comment by jsuchome** [comment]
+1

**Comment by dirkmueller** [comment]
+1

**Pull request merged**